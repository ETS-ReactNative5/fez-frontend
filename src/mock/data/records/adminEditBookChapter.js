export default {
    "rek_pid": "UQ:347671",
    "rek_title_xsdmf_id": 10623,
    "rek_title": "Turing's test: a philosophical and historical guide",
    "rek_description_xsdmf_id": null,
    "rek_description": null,
    "rek_display_type_xsdmf_id": 3673,
    "rek_display_type": 177,
    "rek_status_xsdmf_id": 3680,
    "rek_status": 2,
    "rek_date_xsdmf_id": 6260,
    "rek_date": "2009-01-01T00:00:00Z",
    "rek_object_type_xsdmf_id": 3674,
    "rek_object_type": 3,
    "rek_depositor_xsdmf_id": 7578,
    "rek_depositor": 4059,
    "rek_created_date_xsdmf_id": 3677,
    "rek_created_date": "2014-12-18T13:13:13Z",
    "rek_updated_date_xsdmf_id": 3678,
    "rek_updated_date": "2019-04-30T20:31:04Z",
    "rek_file_downloads": 0,
    "rek_citation": "",
    "rek_genre_xsdmf_id": 7201,
    "rek_genre": "Book Chapter",
    "rek_genre_type_xsdmf_id": null,
    "rek_genre_type": null,
    "rek_formatted_title_xsdmf_id": null,
    "rek_formatted_title": null,
    "rek_formatted_abstract_xsdmf_id": 11491,
    "rek_formatted_abstract": "We set the Turing Test in the historical context of the development of machine intelligence, describe the different forms of the test and its rationale, and counter common misinterpretations and objections. Recently published material by Turing casts fresh light on his thinking.<br />",
    "rek_depositor_affiliation_xsdmf_id": 11881,
    "rek_depositor_affiliation": 924,
    "rek_thomson_citation_count": null,
    "rek_thomson_citation_count_xsdmf_id": null,
    "rek_subtype_xsdmf_id": 12378,
    "rek_subtype": "Research book chapter (original research)",
    "rek_scopus_citation_count": 10,
    "rek_herdc_notes_xsdmf_id": null,
    "rek_herdc_notes": null,
    "rek_scopus_doc_type_xsdmf_id": 12550,
    "rek_scopus_doc_type": "ch",
    "rek_wok_doc_type_xsdmf_id": null,
    "rek_wok_doc_type": null,
    "rek_pubmed_doc_type_xsdmf_id": null,
    "rek_pubmed_doc_type": null,
    "rek_security_inherited": 1,
    "rek_altmetric_score": 0,
    "rek_altmetric_score_xsdmf_id": null,
    "rek_altmetric_id": 10836134,
    "rek_altmetric_id_xsdmf_id": null,
    "rek_copyright_xsdmf_id": 3679,
    "rek_copyright": "on",
    "rek_security_policy": 1,
    "rek_datastream_policy": null,
    "fez_record_search_key_advisory_statement": null,
    "fez_record_search_key_article_number": null,
    "fez_record_search_key_assigned_group_id": [],
    "fez_record_search_key_assigned_user_id": [],
    "fez_record_search_key_author": [
        {
            "rek_author_id": 29021894,
            "rek_author_pid": "UQ:347671",
            "rek_author_xsdmf_id": null,
            "rek_author": "Copeland, Jack",
            "rek_author_order": 1
        },
        {
            "rek_author_id": 29021895,
            "rek_author_pid": "UQ:347671",
            "rek_author_xsdmf_id": null,
            "rek_author": "Proudfoot, Diane",
            "rek_author_order": 2
        }
    ],
    "fez_record_search_key_author_affiliation_country": [],
    "fez_record_search_key_author_affiliation_full_address": [],
    "fez_record_search_key_author_affiliation_id": [],
    "fez_record_search_key_author_affiliation_name": [],
    "fez_record_search_key_author_affiliation_type": [],
    "fez_record_search_key_author_id": [
        {
            "rek_author_id_id": 28389935,
            "rek_author_id_pid": "UQ:347671",
            "rek_author_id_xsdmf_id": null,
            "rek_author_id": 96220,
            "rek_author_id_order": 1,
            "rek_author_id_lookup": "Jack Copeland"
        },
        {
            "rek_author_id_id": 28389936,
            "rek_author_id_pid": "UQ:347671",
            "rek_author_id_xsdmf_id": null,
            "rek_author_id": 96317,
            "rek_author_id_order": 2,
            "rek_author_id_lookup": "Diane Proudfoot"
        }
    ],
    "fez_record_search_key_biosis_id": null,
    "fez_record_search_key_book_title": {
        "rek_book_title_id": 292756,
        "rek_book_title_pid": "UQ:347671",
        "rek_book_title_xsdmf_id": null,
        "rek_book_title": "Parsing the Turing Test: Philosophical and Methodological Issues in the Quest for the Thinking Computer"
    },
    "fez_record_search_key_chapter_number": {
        "rek_chapter_number_id": 137441,
        "rek_chapter_number_pid": "UQ:347671",
        "rek_chapter_number_xsdmf_id": null,
        "rek_chapter_number": "9"
    },
    "fez_record_search_key_content_indicator": [],
    "fez_record_search_key_contributor": [
        {
            "rek_contributor_id": 3113312,
            "rek_contributor_pid": "UQ:347671",
            "rek_contributor_xsdmf_id": null,
            "rek_contributor": "Robert Epstein",
            "rek_contributor_order": 1
        },
        {
            "rek_contributor_id": 3113313,
            "rek_contributor_pid": "UQ:347671",
            "rek_contributor_xsdmf_id": null,
            "rek_contributor": "Gary Roberts",
            "rek_contributor_order": 2
        },
        {
            "rek_contributor_id": 3113314,
            "rek_contributor_pid": "UQ:347671",
            "rek_contributor_xsdmf_id": null,
            "rek_contributor": "Grace Beber",
            "rek_contributor_order": 3
        }
    ],
    "fez_record_search_key_contributor_id": [
        {
            "rek_contributor_id_id": 2322552,
            "rek_contributor_id_pid": "UQ:347671",
            "rek_contributor_id_xsdmf_id": null,
            "rek_contributor_id": 0,
            "rek_contributor_id_order": 1
        },
        {
            "rek_contributor_id_id": 2322553,
            "rek_contributor_id_pid": "UQ:347671",
            "rek_contributor_id_xsdmf_id": null,
            "rek_contributor_id": 0,
            "rek_contributor_id_order": 2
        },
        {
            "rek_contributor_id_id": 2322554,
            "rek_contributor_id_pid": "UQ:347671",
            "rek_contributor_id_xsdmf_id": null,
            "rek_contributor_id": 0,
            "rek_contributor_id_order": 3
        }
    ],
    "fez_record_search_key_corresponding_country": [],
    "fez_record_search_key_corresponding_email": [],
    "fez_record_search_key_corresponding_name": [],
    "fez_record_search_key_corresponding_organisation": [],
    "fez_record_search_key_creator_contribution_statement": [],
    "fez_record_search_key_datastream_policy": null,
    "fez_record_search_key_date_available": null,
    "fez_record_search_key_doi": {
        "rek_doi_id": 1541543,
        "rek_doi_pid": "UQ:347671",
        "rek_doi_xsdmf_id": null,
        "rek_doi": "10.1007/978-1-4020-6710-5_9",
        "fez_altmetric": {
            "as_id": 56280,
            "as_amid": 10836134,
            "as_doi": "10.1007/978-1-4020-6710-5_9",
            "as_score": 0,
            "as_created": "1472541988",
            "as_last_checked": "1563690193",
            "as_1d": 0,
            "as_2d": 0,
            "as_3d": 0,
            "as_4d": 0,
            "as_5d": 0,
            "as_6d": 0,
            "as_1w": 0,
            "as_1m": 0,
            "as_3m": 0,
            "as_6m": 0,
            "as_1y": 0,
            "as_total_posts_count": 0,
            "as_facebook_posts_count": 0,
            "as_policy_posts_count": 0,
            "as_blogs_posts_count": 0,
            "as_googleplus_posts_count": 0,
            "as_news_posts_count": 0,
            "as_reddit_posts_count": 0,
            "as_twitter_posts_count": 0,
            "as_syllabi_posts_count": 0,
            "as_video_posts_count": 0,
            "as_weibo_posts_count": 0,
            "as_qa_posts_count": 0,
            "as_f1000_posts_count": 0,
            "as_wikipedia_posts_count": 0,
            "as_pinterest_posts_count": 0,
            "as_linkedin_posts_count": 0,
            "as_peer_reviews_posts_count": 0,
            "as_citation_url": "http://www.altmetric.com/details.php?citation_id=10836134"
        }
    },
    "fez_record_search_key_edition": null,
    "fez_record_search_key_end_page": {
        "rek_end_page_id": 5533031,
        "rek_end_page_pid": "UQ:347671",
        "rek_end_page_xsdmf_id": null,
        "rek_end_page": "138"
    },
    "fez_record_search_key_file_attachment_access_condition": [],
    "fez_record_search_key_file_attachment_embargo_date": [],
    "fez_record_search_key_file_attachment_name": [],
    "fez_record_search_key_grant_acronym": [],
    "fez_record_search_key_grant_agency": [],
    "fez_record_search_key_grant_agency_id": [],
    "fez_record_search_key_grant_agency_type": [],
    "fez_record_search_key_grant_id": [],
    "fez_record_search_key_grant_text": [],
    "fez_record_search_key_grant_type": [],
    "fez_record_search_key_herdc_code": {
        "rek_herdc_code_id": 4691917,
        "rek_herdc_code_pid": "UQ:347671",
        "rek_herdc_code_xsdmf_id": null,
        "rek_herdc_code": 450006,
        "rek_herdc_code_lookup": "B1"
    },
    "fez_record_search_key_herdc_status": {
        "rek_herdc_status_id": 3601004,
        "rek_herdc_status_pid": "UQ:347671",
        "rek_herdc_status_xsdmf_id": null,
        "rek_herdc_status": 453220,
        "rek_herdc_status_lookup": "Provisional Code"
    },
    "fez_record_search_key_institutional_status": {
        "rek_institutional_status_id": 3260561,
        "rek_institutional_status_pid": "UQ:347671",
        "rek_institutional_status_xsdmf_id": null,
        "rek_institutional_status": 453224,
        "rek_institutional_status_lookup": "Non-UQ"
    },
    "fez_record_search_key_isbn": [
        {
            "rek_isbn_id": 1066448,
            "rek_isbn_pid": "UQ:347671",
            "rek_isbn_xsdmf_id": null,
            "rek_isbn": "9781402067082",
            "rek_isbn_order": 1
        },
        {
            "rek_isbn_id": 1066449,
            "rek_isbn_pid": "UQ:347671",
            "rek_isbn_xsdmf_id": null,
            "rek_isbn": "9781402096242",
            "rek_isbn_order": 2
        },
        {
            "rek_isbn_id": 1066450,
            "rek_isbn_pid": "UQ:347671",
            "rek_isbn_xsdmf_id": null,
            "rek_isbn": "9781402067105",
            "rek_isbn_order": 3
        }
    ],
    "fez_record_search_key_isderivationof": [],
    "fez_record_search_key_isi_loc": null,
    "fez_record_search_key_ismemberof": [
        {
            "rek_ismemberof_id": 11593636,
            "rek_ismemberof_pid": "UQ:347671",
            "rek_ismemberof_xsdmf_id": null,
            "rek_ismemberof": "UQ:3810",
            "rek_ismemberof_order": 1,
            "parent": {
                "rek_pid": "UQ:3810",
                "rek_security_policy": 5,
                "rek_datastream_policy": 5
            },
            "rek_ismemberof_lookup": "School of Historical and Philosophical Inquiry"
        }
    ],
    "fez_record_search_key_ismn": [],
    "fez_record_search_key_issn": [],
    "fez_record_search_key_keywords": [
        {
            "rek_keywords_id": 29276438,
            "rek_keywords_pid": "UQ:347671",
            "rek_keywords_xsdmf_id": null,
            "rek_keywords": "Artificial intelligence",
            "rek_keywords_order": 1
        },
        {
            "rek_keywords_id": 29276439,
            "rek_keywords_pid": "UQ:347671",
            "rek_keywords_xsdmf_id": null,
            "rek_keywords": "Automatic computing engine",
            "rek_keywords_order": 2
        },
        {
            "rek_keywords_id": 29276440,
            "rek_keywords_pid": "UQ:347671",
            "rek_keywords_xsdmf_id": null,
            "rek_keywords": "Bombe",
            "rek_keywords_order": 3
        },
        {
            "rek_keywords_id": 29276441,
            "rek_keywords_pid": "UQ:347671",
            "rek_keywords_xsdmf_id": null,
            "rek_keywords": "Chinese room",
            "rek_keywords_order": 4
        },
        {
            "rek_keywords_id": 29276442,
            "rek_keywords_pid": "UQ:347671",
            "rek_keywords_xsdmf_id": null,
            "rek_keywords": "Cryptanalysis",
            "rek_keywords_order": 5
        },
        {
            "rek_keywords_id": 29276443,
            "rek_keywords_pid": "UQ:347671",
            "rek_keywords_xsdmf_id": null,
            "rek_keywords": "Enigma",
            "rek_keywords_order": 6
        },
        {
            "rek_keywords_id": 29276444,
            "rek_keywords_pid": "UQ:347671",
            "rek_keywords_xsdmf_id": null,
            "rek_keywords": "Intelligence",
            "rek_keywords_order": 7
        },
        {
            "rek_keywords_id": 29276445,
            "rek_keywords_pid": "UQ:347671",
            "rek_keywords_xsdmf_id": null,
            "rek_keywords": "Turing",
            "rek_keywords_order": 8
        },
        {
            "rek_keywords_id": 29276446,
            "rek_keywords_pid": "UQ:347671",
            "rek_keywords_xsdmf_id": null,
            "rek_keywords": "Turing machine",
            "rek_keywords_order": 9
        },
        {
            "rek_keywords_id": 29276447,
            "rek_keywords_pid": "UQ:347671",
            "rek_keywords_xsdmf_id": null,
            "rek_keywords": "Turing test",
            "rek_keywords_order": 10
        }
    ],
    "fez_record_search_key_language": [
        {
            "rek_language_id": 5260969,
            "rek_language_pid": "UQ:347671",
            "rek_language_xsdmf_id": null,
            "rek_language": "eng",
            "rek_language_order": 1
        }
    ],
    "fez_record_search_key_language_of_book_title": [],
    "fez_record_search_key_language_of_title": [],
    "fez_record_search_key_link": [
        {
            "rek_link_id": 3198422,
            "rek_link_pid": "UQ:347671",
            "rek_link_xsdmf_id": null,
            "rek_link": "http://dx.doi.org/10.1007/978-1-4020-6710-5",
            "rek_link_order": 1
        }
    ],
    "fez_record_search_key_link_description": [
        {
            "rek_link_description_id": 3198391,
            "rek_link_description_pid": "UQ:347671",
            "rek_link_description_xsdmf_id": null,
            "rek_link_description": "Full Text from Publisher",
            "rek_link_description_order": 1
        }
    ],
    "fez_record_search_key_native_script_book_title": null,
    "fez_record_search_key_native_script_title": null,
    "fez_record_search_key_notes": null,
    "fez_record_search_key_oa_embargo_days": null,
    "fez_record_search_key_oa_notes": null,
    "fez_record_search_key_oa_status": {
        "rek_oa_status_id": 633692,
        "rek_oa_status_pid": "UQ:347671",
        "rek_oa_status_xsdmf_id": null,
        "rek_oa_status": 453692,
        "rek_oa_status_lookup": "Not yet assessed"
    },
    "fez_record_search_key_place_of_publication": {
        "rek_place_of_publication_id": 4200174,
        "rek_place_of_publication_pid": "UQ:347671",
        "rek_place_of_publication_xsdmf_id": null,
        "rek_place_of_publication": "Berlin, Germany"
    },
    "fez_record_search_key_possible_author_id": [],
    "fez_record_search_key_publisher": {
        "rek_publisher_id": 4465375,
        "rek_publisher_pid": "UQ:347671",
        "rek_publisher_xsdmf_id": null,
        "rek_publisher": "Springer"
    },
    "fez_record_search_key_pubmed_id": null,
    "fez_record_search_key_quality_indicator": [],
    "fez_record_search_key_refereed": null,
    "fez_record_search_key_refereed_source": {
        "rek_refereed_source_id": 1220423,
        "rek_refereed_source_pid": "UQ:347671",
        "rek_refereed_source_xsdmf_id": null,
        "rek_refereed_source": "453638",
        "rek_refereed_source_lookup": "Not yet assessed"
    },
    "fez_record_search_key_roman_script_book_title": null,
    "fez_record_search_key_roman_script_title": null,
    "fez_record_search_key_scopus_id": {
        "rek_scopus_id_id": 2417408,
        "rek_scopus_id_pid": "UQ:347671",
        "rek_scopus_id_xsdmf_id": null,
        "rek_scopus_id": "2-s2.0-80053595592",
        "fez_scopus_citations": null
    },
    "fez_record_search_key_series": null,
    "fez_record_search_key_significance": [],
    "fez_record_search_key_start_page": {
        "rek_start_page_id": 5603412,
        "rek_start_page_pid": "UQ:347671",
        "rek_start_page_xsdmf_id": null,
        "rek_start_page": "119"
    },
    "fez_record_search_key_subject": [],
    "fez_record_search_key_total_chapters": {
        "rek_total_chapters_id": 191592,
        "rek_total_chapters_pid": "UQ:347671",
        "rek_total_chapters_xsdmf_id": null,
        "rek_total_chapters": "29"
    },
    "fez_record_search_key_total_pages": {
        "rek_total_pages_id": 5516475,
        "rek_total_pages_pid": "UQ:347671",
        "rek_total_pages_xsdmf_id": null,
        "rek_total_pages": "20"
    },
    "fez_record_search_key_translated_book_title": null,
    "fez_record_search_key_translated_title": null,
    "fez_record_search_key_volume_number": null,
    "fez_record_search_key_wok_doc_types": [],
    "fez_record_search_key_zoorec_id": null,
    "fez_datastream_info": [
        {
            "dsi_pid": "UQ:347671",
            "dsi_dsid": "FezACML_UQ_347671.xml",
            "dsi_embargo_date": null,
            "dsi_open_access": null,
            "dsi_label": "FezACML security for PID - UQ:347671",
            "dsi_mimetype": "text/xml",
            "dsi_copyright": null,
            "dsi_state": "A",
            "dsi_size": 3705,
            "dsi_security_policy": 1,
            "dsi_security_inherited": 0
        }
    ],
    "fez_matched_journals": [],
    "rek_display_type_lookup": "Book Chapter",
    "rek_pubmed_doc_type_lookup": null,
    "rek_object_type_lookup": "Record",
    "rek_scopus_doc_type_lookup": null,
    "rek_status_lookup": "Published",
    "rek_wok_doc_type_lookup": null,
    "fez_record_search_key_security_policy": [
        {
            "rek_security_policy_id": 193396,
            "rek_security_policy_pid": "UQ:347671",
            "rek_security_policy": 5,
            "rek_security_policy_order": 1
        }
    ]
}