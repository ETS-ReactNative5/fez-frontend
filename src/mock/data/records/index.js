export { default as publicationTypeListAudio } from './publicationTypeListAudio';
export { default as publicationTypeListBook } from './publicationTypeListBook';
export { default as publicationTypeListBookChapter } from './publicationTypeListBookChapter';
export { default as publicationTypeListConferencePaper } from './publicationTypeListConferencePaper';
export { default as publicationTypeListConferenceProceedings } from './publicationTypeListConferenceProceedings';
export { default as publicationTypeListCreativeWork } from './publicationTypeListCreativeWork';
export { default as publicationTypeListDepartmentTechnicalReport } from './publicationTypeListDepartmentTechnicalReport';
export { default as publicationTypeListImage } from './publicationTypeListImage';
export { default as publicationTypeListJournal } from './publicationTypeListJournal';
export { default as publicationTypeListManuscript } from './publicationTypeListManuscript';
export { default as publicationTypeListNewspaperArticle } from './publicationTypeListNewspaperArticle';
export { default as publicationTypeListPatent } from './publicationTypeListPatent';
export { default as publicationTypeListPreprint } from './publicationTypeListPreprint';
export { default as publicationTypeListResearchReport } from './publicationTypeListResearchReport';
export { default as publicationTypeListSeminarPaper } from './publicationTypeListSeminarPaper';
export { default as publicationTypeListThesis } from './publicationTypeListThesis';
export { default as publicationTypeListVideo } from './publicationTypeListVideo';
export { default as publicationTypeListWorkingPaper } from './publicationTypeListWorkingPaper';
export { default as collectionRecord } from './collectionRecord';
export { default as collections } from './collections';
export { default as collectionsByCommunity } from './collectionsByCommunity';
export { default as communityRecord } from './communityRecord';
export { default as communitySearchList } from './communitySearchList';
export { default as externalDoiSearchResultList } from './externalDoiSearchResultList';
export { default as externalPubMedSearchResultsList } from './externalPubMedSearchResultsList';
export { default as externalTitleScopusResultsList } from './externalTitleScopusResultsList';
export { default as externalTitleSearchResultsList } from './externalTitleSearchResultsList';
export { default as incompleteNTROlist } from './incompleteNTROlist';
export { default as incompleteNTROrecord } from './incompleteNTROrecord';
export { default as incompleteNTROrecordUqrdav10 } from './incompleteNTRORecordUqrdav10';
export { default as incompleteNTROrecordUqsbutl1 } from './incompleteNTRORecordUqsbutl1';
export { default as internalTitleSearchList } from './internalTitleSearchList';
export { default as internalTitleSearchListNoResults } from './internalTitleSearchListNoResults';
export { default as lookupToolIncites } from './lookupToolIncites';
export { default as MyDatasetList } from './MyDatasetList';
export { default as myRecordsList } from './myRecordsList';
export { default as possibleUnclaimedList } from './possibleUnclaimedList';
export { default as record } from './record';
export { default as recordsTypeList } from './recordsTypeList';
export { default as recordWithDatastreams } from './recordWithDatastreams';
export { default as recordWithoutAuthorIds } from './recordWithoutAuthorIds';
export { default as recordWithTiffAndThumbnail } from './recordWithTiffAndThumbnail';
export { default as UQ352045 } from './incompleteNTRORecordUQ352045';
export { default as UQ716942uqagrinb } from './incompleteNTRORecordUQ716942_uqagrinb';
export { default as UQ716942uqagrinbGrants } from './incompleteNTRORecordUQ716942_uqagrinb_grants';
