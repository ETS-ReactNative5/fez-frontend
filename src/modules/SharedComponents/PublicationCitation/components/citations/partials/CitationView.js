import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class CitationView extends PureComponent {
    static propTypes = {
        prefix: PropTypes.string,
        suffix: PropTypes.string,
        className: PropTypes.string,
        value: PropTypes.string,
        citationId: PropTypes.string,
    };

    static defaultProps = {
        prefix: ' ',
        suffix: '.',
    };

    constructor(props) {
        super(props);
    }

    render() {
        const { value, className, prefix, suffix, citationId } = this.props;
        if (!value) {
            return <span className={`${className || ''} empty`} data-testid={citationId} />;
        }
        return (
            <span className={className || ''} data-testid={citationId}>
                {prefix}
                {value}
                {suffix === value.slice(-1) ? '' : suffix}
            </span>
        );
    }
}
