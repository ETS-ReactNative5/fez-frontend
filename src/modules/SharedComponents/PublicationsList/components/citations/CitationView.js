import React from 'react';
import PropTypes from 'prop-types';

export default class CitationView extends React.Component {
    static propTypes = {
        prefix: PropTypes.string,
        suffix: PropTypes.string,
        citationClass: PropTypes.string.isRequired,
        children: PropTypes.any
    };

    static defaultProps = {
        prefix: ' ',
        suffix: '.',
        citationClass: ''
    };

    constructor(props) {
        super(props);
    }

    render() {
        const {prefix, children, suffix, citationClass} = this.props;
        return (
            <span className={citationClass}>
                {prefix}{children}{suffix}
            </span>
        );
    }
}
