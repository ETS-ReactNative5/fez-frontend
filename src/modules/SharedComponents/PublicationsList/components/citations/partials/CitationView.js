import React from 'react';
import PropTypes from 'prop-types';

const CitationView = ({className, prefix = ' ', suffix = '.', value}) => {
    if (!value) return (<span className={`${className || ''} empty`} />);
    return (
        <span className={className || ''}>
            {prefix}{value}{suffix === value.slice(-1) ? '' : suffix}
        </span>
    );
};

CitationView.propTypes = {
    prefix: PropTypes.string,
    suffix: PropTypes.string,
    className: PropTypes.string,
    value: PropTypes.string
};

export default CitationView;
