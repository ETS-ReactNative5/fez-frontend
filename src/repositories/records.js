import {post, patch} from './generic';
import * as routes from './routes';

/**
 * Posts a records item
 * @param {object} data to be posted, refer to backend API
 * @returns {Promise}
 */
export function postRecord(data) {
    if (data.rek_pid) return Promise.resolve(data);
    return post(routes.NEW_RECORD_API(), data);
}

/**
 * Patches a records item
 * @param {string} pid of object to be updated
 * @param {object} data to be posted, refer to backend API
 * @returns {Promise}
 */
export function patchRecord(pid, data) {
    return patch(routes.EXISTING_RECORD_API({pid: pid}), data);
}
