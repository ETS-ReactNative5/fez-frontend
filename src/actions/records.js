import {post, patch} from 'repositories/generic';
import * as routes from 'repositories/routes';
import {putUploadFiles} from '../repositories';
import * as transformers from './transformers';
import {NEW_RECORD_DEFAULT_VALUES} from 'config/general';
import * as actions from './actionTypes';

/**
 * Save a new record involves up to three steps: create a new record, upload files, update record with uploaded files.
 * If error occurs on any stage failed action is dispatched
 * @param {object} data to be posted, refer to backend API
 * @returns {promise} - this method is used by redux form onSubmit which requires Promise resolve/reject as a return
 */
export function createNewRecord(data) {
    return dispatch => {
        dispatch({type: actions.CREATE_RECORD_SAVING});

        // set default values, links
        const recordRequest = {
            ...NEW_RECORD_DEFAULT_VALUES,
            ...JSON.parse(JSON.stringify(data)),
            ...transformers.getRecordLinkSearchKey(data),
            ...transformers.getRecordAuthorsSearchKey(data.authors || data.currentAuthor),
            ...transformers.getRecordAuthorsIdSearchKey(data.authors || data.currentAuthor),
            ...transformers.getRecordContributorsSearchKey(data.editors),
            ...transformers.getRecordContributorsIdSearchKey(data.editors),
            ...transformers.getRecordSupervisorsSearchKey(data.supervisors),
            ...transformers.getRecordSubjectSearchKey(data.fieldOfResearch)
        };

        // delete extra form values from request object
        if (recordRequest.authors) delete recordRequest.authors;
        if (recordRequest.editors) delete recordRequest.editors;
        if (recordRequest.files) delete recordRequest.files;
        if (recordRequest.currentAuthor) delete recordRequest.currentAuthor;
        if (recordRequest.supervisors) delete recordRequest.supervisors;
        if (recordRequest.fieldOfResearch) delete recordRequest.fieldOfResearch;

        let newRecord = null;
        const hasFilesToUpload = data.files && data.files.queue && data.files.queue.length > 0;
        const recordPatch = hasFilesToUpload ? {...transformers.getRecordFileAttachmentSearchKey(data.files.queue)} : null;

        return post(routes.NEW_RECORD_API(), recordRequest)
            .then(response => {
                // save new record response
                newRecord = response.data;
                return response;
            })
            .then(() =>(hasFilesToUpload ? putUploadFiles(newRecord.rek_pid, data.files.queue, dispatch) : newRecord))
            .then(() => (hasFilesToUpload ? patch(routes.EXISTING_RECORD_API({pid: newRecord.rek_pid}), recordPatch) : newRecord))
            .then((response) => {
                dispatch({
                    type: actions.CREATE_RECORD_SUCCESS,
                    payload: response.data ? response.data : newRecord
                });
                return Promise.resolve(response.data ? response.data : newRecord);
            })
            .catch(error => {
                // record was created, but file upload or record patch failed
                if (!!newRecord && !!newRecord.rek_pid) {
                    dispatch({
                        type: actions.CREATE_RECORD_SUCCESS,
                        payload: {
                            newRecord: newRecord,
                            fileUploadFailed: true
                        }
                    });

                    return Promise.resolve(newRecord);
                }

                dispatch({
                    type: actions.CREATE_RECORD_FAILED,
                    payload: error.message
                });

                return Promise.reject(error);
            });
    };
}


/**
 * Submit thesis involves two steps: upload files, create record with uploaded files.
 * If error occurs on any stage failed action is dispatched
 * @param {object} data to be posted, refer to backend API
 * @returns {promise} - this method is used by redux form onSubmit which requires Promise resolve/reject as a return
 */
export function submitThesis(data, author) {
    return dispatch => {
        const hasFilesToUpload = data.files && data.files.queue && data.files.queue.length > 0;
        if (!hasFilesToUpload) {
            // reject thesis submission, files are required
            return Promise.reject('Please attach files to proceed with thesis submission');
        }
        // set default values, links
        const recordRequest = {
            ...JSON.parse(JSON.stringify(data)),
            ...transformers.getRecordAuthorsSearchKey(data.currentAuthor),
            ...transformers.getRecordAuthorsIdSearchKey(data.currentAuthor),
            ...transformers.getRecordSupervisorsSearchKey(data.supervisors),
            ...transformers.getRecordSubjectSearchKey(data.fieldOfResearch),
            ...transformers.getRecordFileAttachmentSearchKey(data.files.queue)
        };

        // delete extra form values from request object
        if (recordRequest.authors) delete recordRequest.authors;
        if (recordRequest.editors) delete recordRequest.editors;
        if (recordRequest.files) delete recordRequest.files;
        if (recordRequest.currentAuthor) delete recordRequest.currentAuthor;
        if (recordRequest.supervisors) delete recordRequest.supervisors;
        if (recordRequest.fieldOfResearch) delete recordRequest.fieldOfResearch;

        let fileUploadSucceeded = false;
        dispatch({type: actions.CREATE_RECORD_SAVING});

        // debugger;
        return putUploadFiles(`UQ:${author.aut_student_username}`, data.files.queue, dispatch)
            .then((response) => {
                fileUploadSucceeded = !!response;
                return post(routes.NEW_RECORD_API(), recordRequest);
            })
            .then(response => {
                dispatch({
                    type: actions.CREATE_RECORD_SUCCESS,
                    payload: {
                        newRecord: response
                    }
                });
                return response;
            })
            .catch(error => {
                const specificError = !fileUploadSucceeded
                    ? 'File upload failed. Issue has been created to notify eSpace administrators. '
                    : 'Error occurred while saving record to eSpace. ';
                const compositeError = `${specificError} ${ error.message ? `(${error.message})` : '' }`;

                dispatch({
                    type: actions.CREATE_RECORD_FAILED,
                    payload: compositeError
                });

                return Promise.reject(compositeError);
            });
    };
}
/**
 * Clear new record
 * @returns {action}
 */
export function clearNewRecord() {
    return dispatch => {
        dispatch({
            type: actions.CREATE_RECORD_RESET
        });
    };
}
