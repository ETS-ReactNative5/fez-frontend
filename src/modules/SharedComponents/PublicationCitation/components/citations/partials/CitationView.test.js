import CitationView from './CitationView';
import {researchReport} from 'mock/data/testing/records';

function setup(testProps, isShallow = true) {
    const props = {
        ...testProps,
        prefix: testProps.prefix || ' ',
        suffix: testProps.suffix || '.',
        value: testProps.value,
        className: testProps.className || ''
    };
    return getElement(CitationView, props, isShallow);
}

describe('CitationView test ', () => {
    it('should render component with empty span', () => {
        const wrapper = setup({});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render component with correct props', () => {
        const wrapper = setup({prefix: ' ', suffix: ':', className: 'citationClassName', value: 'Some text' });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render component with correct suffix', () => {
        const wrapper = setup({prefix: ' ', suffix: '.', className: 'citationClassName', value: 'Some text.' });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render component with no prefix or suffix', () => {
        const wrapper = setup({prefix: ' ', suffix: '.', className: 'citationClassName' });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render component with no className', () => {
        const wrapper = setup({value: 'Some text.'});
        expect(toJson(wrapper)).toMatchSnapshot();
    })
});
