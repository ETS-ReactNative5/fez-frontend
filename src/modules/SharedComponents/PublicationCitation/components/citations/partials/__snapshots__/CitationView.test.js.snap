// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitationView test  should render component with correct props 1`] = `
<span
  className="citationClassName"
>
   
  Some text
  :
</span>
`;

exports[`CitationView test  should render component with correct suffix 1`] = `
<span
  className="citationClassName"
>
   
  Some text.
</span>
`;

exports[`CitationView test  should render component with empty span 1`] = `
<span
  className=" empty"
/>
`;

exports[`CitationView test  should render component with no className 1`] = `
<span
  className=""
>
   
  Some text.
</span>
`;

exports[`CitationView test  should render component with no prefix or suffix 1`] = `
<span
  className="citationClassName empty"
/>
`;
